### PR TITLE
support get apiserverurl from auto import secret

### PR DESCRIPTION
**Pull request by zhiweiyin318**
story: https://issues.redhat.com/browse/ACM-13412

### CODE REVIEW

File: pkg/controller/autoimport/apiserver_syncer.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	case constants.AutoImportSecretKubeToken:
+		if kubeServer, ok := secret.Data[constants.AutoImportSecretKubeServerKey]; ok {
+			return string(kubeServer), nil
+		}
+		return "", fmt.Errorf("cannot get APIServer URL from secret %s/%s", secret.Namespace, secret.Name)
+
+	case constants.AutoImportSecretRosaConfig:
+		if apiServer, ok := secret.Data[constants.AutoImportSecretRosaConfigAPIURLKey]; ok {
+			return string(apiServer), nil
```
[the remark this answers is not in the document]
Comment (skeeey) [reply pkg/controller/autoimport/apiserver_syncer.go:78]: it‘s not, it's the OCM api url

File: pkg/controller/autoimport/autoimport_syncer.go [hunk 2]
```diff
@@ -123,20 +65,20 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
...
 						autoImportSecret.Namespace, autoImportSecret.Name, constants.AutoImportRetryName),
 				),
-				r.mcRecorder,
+				s.mcRecorder,
 			); err != nil {
 				return reconcile.Result{}, err
 			}
 			// auto import secret invalid, stop retrying
-			reqLogger.V(5).Info("Auto import secret invalid", "managedCluster", managedCluster.Name)
+			klog.Error("Failed to get autoImportRetry. stop retrying: ", managedCluster.Name)
```
Comment (zhujian7) [inline pkg/controller/autoimport/autoimport_syncer.go:81]: should we print the err here?

File: pkg/controller/autoimport/apiserver_syncer.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		if config.URL == apiServerURL {
+			return reconcile.Result{}, nil
+		}
+	}
+
+	clusterCopy.Spec.ManagedClusterClientConfigs = append(clusterCopy.Spec.ManagedClusterClientConfigs,
+		clusterv1.ClientConfig{
+			URL: apiServerURL,
+		})
+
```
Comment (zhujian7) [inline pkg/controller/autoimport/apiserver_syncer.go:51]: do we need to add a log to indicate that the config URL in the spec is added by import controller?

**Comment by zhujian7** [comment]
LGTM, left some nit comments.
/approve

File: pkg/controller/autoimport/controller.go [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+}
+
+// blank assignment to verify that ReconcileAutoImport implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAutoImport{}
+
+// Reconcile the managed cluster auto import secret to import the managed cluster.
+// Once the managed cluster is imported, the auto import secret will be deleted
+func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	managedClusterName := request.Name
+	klog.V(2).Info("Reconciling auto import controller", "managedCluster", managedClusterName)
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controller/autoimport/controller.go]: done

File: pkg/controller/autoimport/autoimport_syncer.go [hunk 5]
```diff
@@ -42,76 +36,24 @@ type ReconcileAutoImport struct {
...
-
-	autoImportSecret, err := r.informerHolder.AutoImportSecretLister.Secrets(managedClusterName).Get(constants.AutoImportSecretName)
-	if errors.IsNotFound(err) {
-		// the auto import secret could have been deleted, do nothing
-		reqLogger.V(5).Info("Auto import secret not found", "managedCluster", managedCluster.Name)
-		return reconcile.Result{}, nil
-	}
-	if err != nil {
-		return reconcile.Result{}, err
+		klog.V(2).Info("Auto import is enabled：", managedCluster.Name)
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controller/autoimport/autoimport_syncer.go]: done

File: pkg/controller/autoimport/autoimport_syncer.go [hunk 6]
```diff
@@ -146,59 +88,59 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
...
-			r.mcRecorder,
+			s.mcRecorder,
 		); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	reqLogger.V(5).Info("Import result", "importError", iErr, "condition", condition,
-		"current", currentRetry, "result", result, "modified", modified)
+	klog.Info("Import ", managedCluster.Name, " result: ", "importError: ", iErr, " condition: ", condition,
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controller/autoimport/autoimport_syncer.go]: done. only output log when meet error

File: pkg/controller/autoimport/autoimport_syncer.go [hunk 7]
```diff
@@ -146,59 +88,59 @@ func (r *ReconcileAutoImport) Reconcile(ctx context.Context, request reconcile.R
...
 					autoImportSecret.Namespace, autoImportSecret.Name, err),
 			),
-			r.mcRecorder,
+			s.mcRecorder,
 		); err != nil {
 			return reconcile.Result{}, err
 		}
 		// auto import secret invalid, stop retrying
-		reqLogger.V(5).Info("Auto import secret invalid", "managedCluster", managedCluster.Name)
+		klog.Error("Failed to generate clients, stop retrying: ", managedCluster.Name)
```
[the remark this answers is not in the document]
Comment (zhiweiyin318) [reply pkg/controller/autoimport/autoimport_syncer.go]: done

File: pkg/controller/autoimport/apiserver_syncer.go (hunk 3 above)
Comment (zhiweiyin318) [reply pkg/controller/autoimport/apiserver_syncer.go:52]: done

**Pull request merged**